identity-did/src/verification/method_ref.rs

```diff
@@ -8,20 +8,27 @@ use identity_core::common::KeyComparable;
 use identity_core::common::Object;
 
 use crate::did::CoreDID;
-use crate::did::CoreDIDUrl;
+use crate::did::DIDUrl;
+use crate::did::DID;
 use crate::verification::VerificationMethod;
 
 /// A reference to a verification method, either a `DID` or embedded `Method`.
 #[derive(Clone, PartialEq, Deserialize, Serialize)]
 #[serde(untagged)]
-pub enum MethodRef<T = Object> {
-  Embed(VerificationMethod<T>),
-  Refer(CoreDIDUrl), // TODO: could this also be a relative DIDUrl, or another enum variant?
+pub enum MethodRef<D = CoreDID, T = Object>
+where
+  D: DID,
+{
+  Embed(VerificationMethod<D, T>),
+  Refer(DIDUrl<D>), // TODO: could this also be a relative DIDUrl, or another enum variant?
```

Comment:
This is a long-standing TODO I added previously not new from this PR.

It's unclear from the specification whether this has to be a full DIDUrl or not since they describe it as a "URL string":

https://www.w3.org/TR/did-core/#referring-to-verification-methods
> If the value of a [verification method](https://www.w3.org/TR/did-core/#dfn-verification-method) property is a [map](https://infra.spec.whatwg.org/#ordered-map), the [verification method](https://www.w3.org/TR/did-core/#dfn-verification-method) has been embedded and its properties can be accessed directly. However, if the value is a URL [string](https://infra.spec.whatwg.org/#string), the [verification method](https://www.w3.org/TR/did-core/#dfn-verification-method) has been included by reference and its properties will need to be retrieved from elsewhere in the [DID document](https://www.w3.org/TR/did-core/#dfn-did-documents) or from another [DID document](https://www.w3.org/TR/did-core/#dfn-did-documents). This is done by dereferencing the URL and searching the resulting [resource](https://www.w3.org/TR/did-core/#dfn-resources) for a [verification method](https://www.w3.org/TR/did-core/#dfn-verification-method) [map](https://infra.spec.whatwg.org/#ordered-map) with an id property whose value matches the URL.

Reply:
To actually address the comment, I would not make it `non_exhaustive` since we exhaustively match on it internally, and it's unlikely that we will allow relative DIDUrls since there are no examples I can find in the specification which use e.g. just the method fragment. This was mostly pondering a space optimisation.

Reply:
Removed the TODO, given all the above we are unlikely to support partial URL fragments as method references.